root/components/RelationshipsTable.js

```diff
@@ -0,0 +1,177 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import EntityLink from '../static/scripts/common/components/EntityLink';
+import loopParity from '../utility/loopParity';
+import generateRelationshipAppearancesList
+  from '../utility/generateRelationshipAppearancesList';
+import ArtistCreditLink
+  from '../static/scripts/common/components/ArtistCreditLink';
+import commaList from '../static/scripts/common/i18n/commaList';
+import formatDatePeriod
+  from '../static/scripts/common/utility/formatDatePeriod';
+import formatTrackLength
+  from '../static/scripts/common/utility/formatTrackLength';
+import displayLinkAttribute
+  from '../static/scripts/common/utility/displayLinkAttribute';
+
+type Props = {|
+  +entity: CoreEntityT,
+  +fallbackMessage?: string,
+  +heading: string,
+  +showCredits?: boolean,
+|};
+
+const RelationshipsTable = ({
+  entity,
+  fallbackMessage,
+  heading,
+  showCredits,
+}: Props) => {
+  const appearances = generateRelationshipAppearancesList(entity);
+  const relationshipTypes = Object.keys(appearances);
+  if (!appearances || relationshipTypes.length === 0) {
+    return <p>{fallbackMessage}</p> || null;
+  }
+  let hasCreditColumn = 0;
+  let hasAttributeColumn = 0;
+  let hasArtistColumn = 0;
+  let hasLengthColumn = 0;
+  for (const relationshipType of relationshipTypes) {
+    for (const relationship of appearances[relationshipType]) {
```

Comment:
Since `appearances` is just `entity.relationships` grouped by type, you should be able to remove the outer loop and just use `entity.relationships` in place of `appearances[relationshipType]` here.

Reply:
Ignore, I forgot most of the pages other than for place load other "non-appearance" relationships for display.